test/distributed/_composable/test_composability/test_pp_composability.py

```diff
@@ -232,16 +235,19 @@ def build_stage(stage_idx, num_stages):
                     parts[0] = str(int(parts[0]) + offset)
                     name = ".".join(parts)
                     ref_p = ref_parameters[name]
+                    # TODO: WITH FSDP
```

Comment:
p.grad remains plain `torch.Tensor` because last_forward=False, see repros in https://github.com/pytorch/pytorch/pull/134375
```
[rank0]:V0823 15:41:06.855000 4139345 torch/distributed/pipelining/stage.py:533] [0] backward_type='full' set_is_last_backward(last_backward=False) set_requires_gradient_sync(last_backward=False) self._seen_bwd_chunks=15, self.chunks=8


[rank3]:V0823 15:41:06.922000 4139348 torch/distributed/pipelining/stage.py:533] [1] backward_type='weight' set_is_last_backward(last_backward=True) set_requires_gradient_sync(last_backward=True) self._seen_bwd_chunks=15, self.chunks=8
```